server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -110,7 +112,7 @@ public void indexValue(@NotNull Map<String, Object> value, IndexDocumentBuilder
                 }
             }
             var innerValue = value.get(innerName);
-            docBuilder.checkColumnConstraint(child.ident(), innerValue);
+            docBuilder.checkColumnConstraint(child.ident(), innerValue, noValue);
```

Comment:
Can't we clean up further? It looks like the `if` block right above is the only place that will ever call `checkColumnConstraint()` or `verify()` methods with `noValue = true`. Can we simplify the whole PR by wrapping these two lines in an `else` block?

Reply:
That was one of my first drafts in https://github.com/crate/crate/commit/8b33266f7f5ebd0049ea4378763fdb96419af28b
but we can't skip the whole check only if value is not provided - we still need to apply CHECK/MultiChecK/NotNull and their are hidden in `checkColumnConstraint`